pkgs/applications/editors/vim/macvim.nix

```
@@ -28,13 +38,13 @@ in
 stdenv.mkDerivation {
   pname = "macvim";
 
-  version = "8.2.3455";
+  version = "r178";
 
   src = fetchFromGitHub {
     owner = "macvim-dev";
     repo = "macvim";
-    rev = "snapshot-172";
-    sha256 = "sha256-LLLQ/V1vyKTuSXzHW3SOlOejZD5AV16NthEdMoTnfko=";
+    rev = "release-178";
```

This should be derived from version

---

Are you proposing that this should use `finalAttrs` and programmatically change `r###` into `release-###`, so someone overriding the version to `r179` will automatically update the rev too? Or do you just not like that the `178` is duplicated? I'm partial to the idea that changing the version changes the rev too, but that requires defining what happens if someone overrides the version to something that doesn't match `r###`.

---

> Are you proposing that this should use `finalAttrs` and programmatically change `r###` into `release-###`, so someone overriding the version to `r179` will automatically update the rev too?

First yes but then I remembered that we are stripping prefixes like r or v from versions, so we just need to substitute it here.

```suggestion
    rev = "release-${version}";
```

---

Done